spec/kolekti_analizo/collector_spec.rb

```diff
@@ -57,10 +62,14 @@
     end
 
     describe 'default_value_from' do
-      let(:metric_list) { FactoryGirl.build(:analizo_metric_collector_list).raw }
+      let!(:metric_list) { FactoryGirl.build(:analizo_metric_collector_list).raw }
```

Comment:
That ship has long sailed: https://github.com/mezuro/kolekti_analizo/commit/563e544080465aa6671cfc6ed9b65a3e165b8cda. We left this slip when accepting this commit probably for the sake of simpleness.

We can't mock protected/private methods so are you asking about refactoring the whole class? Maybe as an issue on Kolekti in order to ensure others do not fall in the same mistake? But still is not clear to me how should this refactoring get done. Maybe a new class MetricParser? Maybe the real issue is calling the parse on initialization and it should be lazy? What about moving the metric list to a yml (probably other collectors, as phpmd, do not have this issue as they do not need mock at all)?